src/index.js

```diff
@@ -183,33 +183,36 @@ class Masq {
           // check challenges
           if (json.challenge !== challenge) {
             // This peer may be malicious, close the connection
+            this._requestMasqAccessErr = Error('Challenge does not match')
```

Comment:
I'm not sure maintaining an error on `this` is a nice thing to do

Reply:
We maintain done and potentially an error on this.
The other solution would be to use events with 'done' and 'error' events on some event emitter.
The user would have to set event handlers and then call request. It might lead to a more complex API.
What do you think ?